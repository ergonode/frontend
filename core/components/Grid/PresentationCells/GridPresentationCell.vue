/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template functional>
    <div class="presentation-cell">
        <span
            :title="props.value"
            v-text="props.value" />
        <span
            v-if="props.value && props.suffix"
            :title="props.suffix"
            v-text="props.suffix" />
    </div>
</template>

<script>

export default {
    name: 'GridPresentationCell',
    props: {
        value: {
            type: [String, Number],
            required: true,
        },
        suffix: {
            type: String,
            default: '',
        },
    },
};
</script>

<style lang="scss" scoped>
    .presentation-cell {
        display: flex;
        flex: 1;
        justify-content: space-between;
        align-items: center;
        width: 0;
        padding: 8px;
        color: $GRAPHITE_DARK;
        font: $FONT_MEDIUM_12_16;
        user-select: none;
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
    }
</style>
