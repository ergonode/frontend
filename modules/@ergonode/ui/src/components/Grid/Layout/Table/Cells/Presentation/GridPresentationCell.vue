/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template functional>
    <span
        class="grid-presentation-cell"
        v-text="props.value" />
</template>

<script>

export default {
    name: 'GridPresentationCell',
    props: {
        /**
         * Component value
         */
        value: {
            type: [
                String,
                Number,
            ],
            required: true,
        },
    },
};
</script>

<style lang="scss" scoped>
    .grid-presentation-cell {
        flex: 1;
        width: 0;
        padding: 8px 0 8px 8px;
        color: $GRAPHITE_DARK;
        font: $FONT_MEDIUM_12_16;
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
    }
</style>
