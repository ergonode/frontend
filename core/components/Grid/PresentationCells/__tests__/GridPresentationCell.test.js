/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import { shallowMount } from '@vue/test-utils';
import GridPresentationCell from '~/core/components/Grid/PresentationCells/GridPresentationCell';

describe('Grid/PresentationCells/GridPresentationCell', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(GridPresentationCell, {
            propsData: {
                value: 'Test',
            },
        });
    });

    it('Component is named well', () => {
        expect(typeof GridPresentationCell.name).toBe('string');
        expect(GridPresentationCell.name).toEqual('GridPresentationCell');
    });

    it('Component is displaying text', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
