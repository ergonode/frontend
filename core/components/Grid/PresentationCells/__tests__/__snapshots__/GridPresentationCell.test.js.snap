// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid/PresentationCells/GridPresentationCell Component is displaying text 1`] = `
<div class="presentation-cell"><span title="Test">Test</span>
  <!---->
</div>
`;
